smallrye-reactive-messaging-kafka/src/main/java/io/smallrye/reactive/messaging/kafka/KafkaCDIEvents.java

```diff
@@ -0,0 +1,97 @@
+package io.smallrye.reactive.messaging.kafka;
+
+import java.lang.annotation.Annotation;
+import java.util.concurrent.CompletionStage;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Event;
+import javax.enterprise.event.NotificationOptions;
+import javax.enterprise.util.TypeLiteral;
+import javax.inject.Inject;
+
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.producer.Producer;
+
+@ApplicationScoped
+public class KafkaCDIEvents {
+    @Inject
+    Event<Consumer<?, ?>> consumerEvent;
+
+    @Inject
+    Event<Producer<?, ?>> producerEvent;
+
+    public Event<Consumer<?, ?>> consumer() {
+        return consumerEvent;
+    }
+
+    public Event<Producer<?, ?>> producer() {
+        return producerEvent;
+    }
+
+    public KafkaCDIEvents() {
+    }
+
+    private KafkaCDIEvents(boolean isNoop) {
```

Comment:
Can you add a comment about this constructor that is not used during "normal" operations.

Reply:
Is it even used?

Reply:
Not anymore. I thought I removed it.. but I may not have pushed the change. Will double check.